tests/driver/tool_driver.ts

```diff
@@ -0,0 +1,140 @@
+import Debug from "debug";
+import * as fs from "fs";
+import path from "path";
+import { SetupSettings } from "tests/driver";
+import { ARGS } from "tests/utils";
+
+import { GenericTrunkDriver } from "./driver";
+
+const baseDebug = Debug("Driver");
+
+let testNum = 1;
+const toolTests = new Map<string, number>();
+
+const getDebugger = (tool?: string) => {
+  if (!tool) {
+    // If a tool is not provided, provide a counter for easy distinction
+    return baseDebug.extend(`test${testNum++}`);
+  }
+  const numToolTests = toolTests.get(tool);
+  const newNum = (numToolTests ?? 0) + 1;
+  toolTests.set(tool, newNum);
+  return baseDebug.extend(tool).extend(`${newNum}`);
+};
+
+/**
+ * The result of running a 'trunk check' or 'trunk fmt' command.
+ */
+export interface TrunkToolRunResult {
+  exitCode: number;
+  stdout: string;
+  stderr: string;
+  /** Error thrown if the trunk invocation returned a nonzero exit code */
+  error?: Error;
+}
+
+export class TrunkToolDriver extends GenericTrunkDriver {
+  /** The name of the linter. If defined, enable the linter during setup. */
+  tool?: string;
+  /** Dictated version to enable based on logic of parsing environment variables. May be a version string or `LinterVersion` */
+  toEnableVersion?: string;
```

Comment:
Is your intention here to reuse `PLUGINS_TEST_LINTER_VERSION`? Might be worth a discussion